utility/servicer/module.go

```diff
@@ -182,31 +356,107 @@ func (s servicer) admitRelay(relay *coreTypes.Relay) error {
 
 	height := s.GetBus().GetConsensusModule().CurrentHeight()
 	if err := s.validateRelayMeta(relay.Meta, int64(height)); err != nil {
-		return fmt.Errorf("%w: %s", errValidateRelayMeta, err.Error())
+		return fmt.Errorf("%s: %w", err.Error(), errValidateRelayMeta)
 	}
 
 	// TODO: update the CLI to include ApplicationAddress(or Application Public Key) in the RelayMeta
-	session, err := s.GetBus().GetUtilityModule().GetSession(relay.Meta.ApplicationAddress, int64(height), relay.Meta.RelayChain.Id, relay.Meta.GeoZone.Id)
+	session, err := s.getSession(relay)
 	if err != nil {
-		return fmt.Errorf("%s: failed to get a session for height %d for relay meta %s: %w", errPrefix, height, relay.Meta, err)
+		return err
 	}
 
 	// TODO: (REFACTOR) use a loop to run all validators: would also remove the need for passing the session around
```

Comment:
Seems like this TODO is no longer relevant? If not, remove.

Reply:
Thanks, fixed.